tools/azure/hpcc-aks-cluster-insights.py

```diff
@@ -0,0 +1,4251 @@
+            print(f"Error running command: {' '.join(command)}", file=sys.stderr)
+            print(f"Error output: {e.stderr}", file=sys.stderr)
+            return {}
+        except json.JSONDecodeError as e:
+            print(f"Error parsing JSON: {e}", file=sys.stderr)
+            return {}
+
+    def set_subscription(self):
+        """Set the active Azure subscription."""
+        print(f"Setting subscription to: {self.subscription}")
+        subprocess.run(
+            ["az", "account", "set", "--subscription", self.subscription],
+            check=True
+        )
+
+    def get_cluster_details(self) -> Dict[str, Any]:
+        """Fetch AKS cluster details."""
+        if not self.resource_group:
+            # Find resource group
+            clusters = self.run_az_command([
+                "az", "aks", "list",
+                "--subscription", self.subscription,
+                "--query", f"[?name=='{self.cluster_name}']",
+                "-o", "json"
+            ])
+            if clusters and len(clusters) > 0:
+                self.resource_group = clusters[0].get("resourceGroup")
+
+        if not self.resource_group:
+            print(f"Error: Could not find resource group for cluster {self.cluster_name}", file=sys.stderr)
+            return {}
+
+        print(f"Fetching cluster details for {self.cluster_name}...")
+        self.cluster_data = self.run_az_command([
+            "az", "aks", "show",
+            "--name", self.cluster_name,
+            "--resource-group", self.resource_group,
+            "-o", "json"
+        ])
+        return self.cluster_data
+
+    def get_node_pool_summary(self) -> List[Dict[str, Any]]:
+        """Summarize node pool information."""
+        if not self.cluster_data:
+            return []
+
+        node_pools = self.cluster_data.get("agentPoolProfiles", [])
+        summary = []
+
+        for pool in node_pools:
+            pool_name = pool.get("name", "").lower()
+
+            # Determine pool purpose based on name patterns
+            if "thor" in pool_name:
+                if "d" in pool_name and any(size in pool_name for size in ["32", "48", "64"]):
+                    thor_cluster = f"thor-{pool_name}"
+                    purpose = "Thor Workers"
+                    # Estimate workers from pool name if possible
+                    if "400" in pool_name or "d48" in pool_name or "d64" in pool_name:
+                        workers = 400
+                    elif "100" in pool_name:
+                        workers = 100
+                    elif "40" in pool_name:
+                        workers = 40
+                    else:
+                        workers = 0  # Unknown
+                    description = f"Thor worker cluster ({workers} workers estimated)" if workers > 0 else "Thor worker cluster"
+                elif "l" in pool_name:
+                    thor_cluster = f"thor-{pool_name}"
+                    purpose = "Thor Spill Workers"
```

Comment:
generally, what does "Thor Spill Workers" mean? (sounds like it might be something specific to prod400 env.)

The tool should be kept agnostic, so we can point at healthcar, fcra etc.. / any env.